### ISSUE TITLE
MW72 meta/iast conversion

### ISSUE THREAD

**Issue by funderburkjim** [sanskrit-lexicon/Cologne#215]
This issue for comments related to the meta-line/iast conversion of mw72.txt, the Cologne digitization of `Monier-Williams Sanskrit-English Dictionary, 1872`.

Note that we have previously done a provisional conversion of the AS (letter-number) representation of Latin alphabet with diacritics to Unicode representation.  However,  for reasons which seemed plausible for correction work being done at the time (late 2016), the resulting IAST differed from modern IAST.  As part of the current conversion, we will attempt to change things so modern IAST is present.

**Comment by gasyoun** [comment]
>  As part of the current conversion, we will attempt to change things so modern IAST is present.

A tough task, as there are those `ri`shis around there.

**Comment by funderburkjim** [comment]
### power outage

We had a big spring snow, with heavy winds.  This resulted in power outage for about 6 days at my house.  Power restored Wednesday afternoon.  So this work can continue.

**Comment by funderburkjim** [comment]
### IAST issues

A reasonably good approximation to modern IAST is now present in the MW72 coding.
The changes from previous non-standard IAST to standard IAST are:

|old|new|
|---|---|
|ṛi | ṛ |
|ṛī | ṝ |
|Ṛi | Ṛ |
|Ṛī | Ṝ |
|lṛi | ḷ |
|lṛī | ḹ |
|ṉ | ṃ |
|ć | c |
|ćh | ch |
|sh | ṣ |
|Sh | Ṣ |
|ṁ | ṃ |
|Ć | C |
|ṅ | ñ |
|ń | ṅ |
|Ń | Ṅ | 

Changing from one diacritic to another is straightforward.   However, the changes from 'sh' to 'ṣ' is trickier, since 'sh' can occur in English words also.   The actual changes from 'sh' were done based on 
a specific listing of all words containing 'sh', and then a weeding out from this list (of about 1800) words
those words identified as English; the 'sh' conversion was then applied to the remaining list of 800 or so Sanskrit words. The end result is probably of high accuracy.

There are also some funky spellings of various types.  The most prominent is the word 'Sanskrit', which 
appears 100+ times, and is shown in the printed text as  'Sanskṛit'  (note r-underdot). Using the table
directly would leave `Sanskṛt`, but this is too weird so the Anglicized form ''Sanskṛit' was used in this case.

There are probably a few other weird hybrid spellings that we may want to adjust as they are noticed.

#### No changes in Preface
The changes mentioned above were NOT applied to the digitization of the Preface;  the reason is that
there are sections in the Preface where the author details his choices and reasons for the printed 
digitization.   Presenting these explanations with modern IAST would be confusing.

Other parts of the Preface mention literary sources;  when we make use of this material  (e.g. to generate
abbreviation tips), we will want to apply modern IAST.

**Comment by funderburkjim** [comment]
### Sections of the digitization

Comments (lines starting with ';') have been added to delineate the parts of the digitization.
```
      3:; TITLE
     23:; PREFACE
     40:; PREFACE: Reasons for undertaking a New Sanskṛit Dictionary.
    198:; PREFACE: Plan and Arrangement of the Present Work.
    520:; PREFACE: Extent of Sanskṛit Literature comprehended.
    669:; PREFACE: Alphabet and System of Transliteration employed.
    989:; PREFACE: Principal Sources
   1214:; PREFACE: Aids and Encouragements received.
   1269:; PREFACE: Defects and Inconsistencies acknowledged.
   1361:; DIRECTIONS TO BE STUDIED BEFORE USING THIS DICTIONARY.
   1416:; ABBREVIATIONS AND SYMBOLS USED IN THE DICTIONARY.
   1647:; NAGARI LETTERS WITH THEIR INDO-ROMANIC EQUIVALENTS
   1731:; ENTRIES
 439217:; ADDITIONS AND CORRECTIONS.
```

Note that the last section of ADDITIONS is also a source of headwords.
For instance the first headword of the additions/corrections is `akaTam`, an addition.
The second such headword is 'akula', which amplifies the entry appearing in the main ENTRIES:

![image](https://user-images.githubusercontent.com/6393033/37230447-896fc902-23b5-11e8-8203-ef3e62d9101c.png)

**Comment by funderburkjim** [comment]
### Two lines of entries

As mentioned in the preface,  Monier-Williams has *two* lines of headwords in this 1872 edition;  compare
this with the *four* lines of the 1899 edition.
The first line shows headwords in Devanagari, and the second shows headwords in IAST.

#### Many 'subheadwords' to be mined.

Notice `Akula-tā` in the image above.  It is clearly a candidate for sub-headword-hood.   There are
lots like that.   Many will be identifiable in the digitization by the pattern `{%—`   (italic text starting with
em-dash).  There are 55,000+ such cases.

**Comment by funderburkjim** [comment]
The installation of the meta/iast conversion of MW72 is now completed at Cologne.

**Comment by gasyoun** [comment]
Let us kill the dot and leave hybrid Sanskrit, @SergeA ?

**Comment by SergeA** [comment]
> Using the table directly would leave Sanskṛt, but this is too weird so the Anglicized form ''Sanskrit' (no dot under r) was used in this case. 

I think the consistency should be in the first place.
MW used digraph "ṛi" for easy reading. But if we change it to IAST, we should do it everywhere. There are many similar cases: Kṛṣṇa, ṛṣi, prakṛti etc. etc. I think _Sanskrit_ shouldn't be treated separately.

**Comment by gasyoun** [comment]
> I think Sanskrit shouldn't be treated separately.

Must agree. Will raise even more issues in the future otherwise.

**Comment by funderburkjim** [comment]
There is one aspect of *Sanskrit* . Suppose we change the spelling to *Sanskṛt*, as in other words.   Then this spelling is still NOT the AS spelling of a Sanskrit word; the Sanskrit word for 'Sanskrit' is *saṃskṛta*.

Note the usage in MW dictionary under saṃ-skṛta
`n. the Sanskrit language (cf. above ) Ṡiksh.  ; Bhar.  ; Daṡar.  &c. [L=227087.5]`

So, I think the situation with *Sanskṛt* is different from Kṛṣṇa, ṛṣi, prakṛti , all of which are valid spellings of Sanskrit words.

In other words the original *Sanskṛit* is not the spelling of a Sanskrit word so it makes no sense to 
represent it as the IAST spelling of a Sanskrit word.   There is no inconsistency in using 'Sanskrit'; there
is inconsistency in using *Sanskṛit* or *Sanskṛt*.

That's the way this detail appears to me currently.

**Comment by funderburkjim** [comment]
Since this change is controversial, I added it as a case in [mw72_printchange.txt](https://github.com/sanskrit-lexicon/CORRECTIONS/blob/master/dictionaries/MW72/mw72_printchange.txt).

**Comment by SergeA** [comment]
Yes, I agree, "Sanskṛit" is not a real Sanskrit word.
But I insist that the position of "Kṛishṇa" is the same!

There are 2 main types of Sanskrit words in the dictionaries:

1. Exact quotation (including sentences from texts and grammatical forms and stems) which we can name **real Sanskrit**. In MW99 is given in italics. E.g. "see _yajur-veda_" is a reference to the Sanskrit word-stem. Real Sankrit words always follow only Sanskrit grammar.

2. Terminological usage of Sanskrit words in English texts (including all special terms and proper names etc.) which we can name **borrowed Sanskrit**. In MW99 is given in normal font. E.g. "the Yajur-veda" is used as a term, denoting a certain text. Borrowed words follow English grammar and are able to take English affixes: e.g. Veda, Vedas, Vedic, pre-Vedic etc. Borrowed words can keep their original Sanskrit spelling **intact** (as "Veda") **or** can be **altered** (as "Ved-ic" when adding suffix drops the final "a" of the stem).

So the difference between "Kṛishṇa" and "Sanskṛit" is that the former is an intact borrowing, while the latter is an altered borrowing (dropping final "a" and nasalization of "n" of its original _saṃskṛita_ under the influence of modern Indian and European usage). But in relation to "ṛi" both "Kṛishṇa" and "Sanskṛit" are similar.

MW considered that spellings with "ṛ" (Kṛshṇa, Sanskṛt) are hard to pronounce for common people. Thus why he decided to spell "ṛi" everywhere. But when we are trying to bring his transliterations to the more strict standard of IAST, the problem of hard-to-pronounce spellings again arises.
